ballerina/engine.bal

```diff
@@ -21,20 +21,40 @@ import graphql.parser;
 isolated class Engine {
     private final readonly & __Schema schema;
     private final int? maxQueryDepth;
+    private final readonly & (readonly & Interceptor)[] interceptors;
+    private int nextInterceptor;
 
-    isolated function init(string schemaString, int? maxQueryDepth, Service s) returns Error? {
+    isolated function init(string schemaString, int? maxQueryDepth, Service s, (readonly & Interceptor)[] interceptors) returns Error? {
         if maxQueryDepth is int && maxQueryDepth < 1 {
             return error Error("Max query depth value must be a positive integer");
         }
         self.maxQueryDepth = maxQueryDepth;
         self.schema = check createSchema(schemaString);
+        self.interceptors = interceptors.cloneReadOnly();
```

Comment:
If we mark the `interceptors` field as readonly, we don't need to use `cloneReadOnly()` here.

Reply:
Updated